lib/fetch-snyk-docker-analyzer.js

```diff
@@ -41,39 +41,30 @@ function fetch() {
         'https://snyk.io/resources/cli/plugins/docker-analyzer/' +
         `${version}/${getBinaryName()}`;
 
-      const fsStream = fs.createWriteStream(localPath + '.part');
-      const req = request(downloadUrl);
-      req
-        .on('response', function (res) {
-          if (res.statusCode >= 400) {
-            var err = new Error(
-              'Bad HTTP response for snyk-docker-analyzer download');
-            err.statusCode = res.statusCode;
-            fsStream.destroy();
-            reject(err);
-            req.abort();
-            return;
-          }
+      console.log(`Downloading ${getBinaryName()} to ` +
+        `${path.dirname(localPath)} ...`);
 
-          console.log(`Downloading ${getBinaryName()} to ` +
-            `${path.dirname(getBinaryLocalPath())} ...`);
-        })
-        .on('error', function (err) {
-          reject(err);
-        })
-        .pipe(fsStream)
-        .on('error', function (err) {
-          fsStream.destroy();
+      needle.get(downloadUrl, function (error, res) {
+        if (error) {
+          reject(error);
+          return;
+        }
+
+        if (res.statusCode >= 400) {
+          var err = new Error(
+            'Bad HTTP response for snyk-docker-analyzer download');
+          err.statusCode = res.statusCode;
+          err.body = res.body;
           reject(err);
-        })
-        .on('finish', function () {
-          // padding log() with some whitespaces to not mix with the CLI spinner
-          console.log('  => download complete.' + ' '.repeat(50));
-          fs.renameSync(localPath + '.part', localPath);
-          const CHMOD_WITH_EXEC = 0755;
-          fs.chmodSync(localPath, CHMOD_WITH_EXEC);
-          resolve(localPath);
-        });
+          return;
+        }
+
+        console.log('Download complete!');
```

Comment:
the original 

```javascript
console.log('  => download complete.' + ' '.repeat(50));
```

was used to play nicer with the CLI spinner - maybe leave it as it was?

Reply:
Yes! Done.